Makefile

```diff
@@ -5,4 +5,4 @@ HEADERS = -I"src/main/headers" -I"${JAVA_HOME}\include" -I"${JAVA_HOME}\include\
 all: install
 
 install:
-	gcc -Wall -D_JNI_IMPLEMENTATION_ -Wl,--kill-at $(HEADERS) -shared -osrc/main/resources/integrations.dll src/main/native/*.cpp -lcrypt32
\ No newline at end of file
+	g++ -Wall -D_JNI_IMPLEMENTATION_ -Wl,--kill-at $(HEADERS) -shared -osrc/main/resources/integrations.dll src/main/native/*.cpp -lcrypt32 -lshell32 -lole32 -luuid
```

Comment:
1. Are all these linked libs really required? UUID?
2. Have you made sure they are guaranteed to be available on every Windows installation?

Reply:
> Are all these linked libs really required? UUID?

Yes, see https://bugzilla.mozilla.org/show_bug.cgi?id=529327.

> Have you made sure they are guaranteed to be available on every Windows installation?

For two of the three added libs it is easy:
* `ole32.dll` is a core part of Window's [COM library](https://docs.microsoft.com/en-us/windows/win32/com/the-com-library)
* `shell32.dll` is one of the core libraries for the Windows [shell](https://docs.microsoft.com/en-us/windows/win32/shell/versions)

About `uuid`: It seems to be part of the windows 10 SDK (specifically, the file is called `uuid.lib`) but i cannot find any specific info about it nor if it works on systems without the SDK.

Reply:
Would running this on older win version result in the lib not being loadable? Or what are the expected symptoms?

Reply:
> older win version

If you mean with older versions anything before Windows 10, is this actually a goal? I thought we only support windows 10 and everything else is a bonus. Nonetheless, this needs to be tested on a fresh and clean windows 10 installation.

Reply:
So far i found out the following: To build our native functions the [Windows 10 SDK](https://developer.microsoft.com/en-US/windows/downloads/windows-10-sdk/) needs to be installed. But only for building, once this is done, it is no longer a requirement.

Reply:
Yes, older versions are nice-to-have. But if we can avoid incompatibilities, it is at least worth trying. So if that uuid lib  is only required during compile time, thats fine. But maybe there is a better way to tell this the linker? I'm no expert on this field, but as far as I know the `-l` flags tell it that the program can rely on this library being present during runtime.

Reply:
From my point of view it is like with the JRE and JDK. You use the Windows 10 SDK to develop applications, which run on every windows 10 system without additional libs. So all linked libs are present in some dll on Windows 10, but i really cannot say which one (i'm unable to find any doc about this uuid lib, only that it sometimes causes compile problems because it is located in the SDK...)

I did an integration test on a different machine without the SDK and it worked. But that's just a single test case.